EKF/estimator_interface.cpp

```diff
@@ -512,34 +512,46 @@ void EstimatorInterface::setDragData(const imuSample &imu)
 bool EstimatorInterface::initialise_interface(uint64_t timestamp)
 {
 	// find the maximum time delay the buffers are required to handle
-	const uint16_t max_time_delay_ms = math::max(_params.mag_delay_ms,
-					     math::max(_params.range_delay_ms,
-					       math::max(_params.gps_delay_ms,
-						 math::max(_params.flow_delay_ms,
-						   math::max(_params.ev_delay_ms,
-						     math::max(_params.auxvel_delay_ms,
-						       math::max(_params.min_delay_ms,
-							 math::max(_params.airspeed_delay_ms,
-							 	     _params.baro_delay_ms))))))));
+	float max_time_delay_ms = math::max(_params.baro_delay_ms,
+					math::max(_params.auxvel_delay_ms,
+						_params.airspeed_delay_ms));
+
+	// mag mode
+	if (_params.mag_fusion_type != MAG_FUSE_TYPE_NONE) {
+		max_time_delay_ms = math::max(_params.mag_delay_ms, max_time_delay_ms);
+	}
+
+	// range aid or range height
+	if (_params.range_aid || (_params.vdist_sensor_type == VDIST_SENSOR_RANGE)) {
+		max_time_delay_ms = math::max(_params.range_delay_ms, max_time_delay_ms);
+	}
+
+	if (_params.fusion_mode & MASK_USE_GPS) {
+		max_time_delay_ms = math::max(_params.gps_delay_ms, max_time_delay_ms);
+	}
+
+	if (_params.fusion_mode & MASK_USE_OF) {
+		max_time_delay_ms = math::max(_params.flow_delay_ms, max_time_delay_ms);
+	}
+
+	if (_params.fusion_mode & (MASK_USE_EVPOS | MASK_USE_EVYAW | MASK_USE_EVVEL)) {
+		max_time_delay_ms = math::max(_params.ev_delay_ms, max_time_delay_ms);
+	}
 
 	// calculate the IMU buffer length required to accomodate the maximum delay with some allowance for jitter
-	_imu_buffer_length = (max_time_delay_ms / FILTER_UPDATE_PERIOD_MS) + 1;
+	_imu_buffer_length = ceilf(max_time_delay_ms / FILTER_UPDATE_PERIOD_MS) + 1;
```

Comment:
Does it need the +1 if you have `ceilf`?

Reply:
Perhaps not, but the tolerances in the unit tests need to be adjusted slightly if it changes.